### PR TITLE
Nano: add auto channels_last_3d support

### PR DESCRIPTION
**Pull request by TheaperDeng**
### 1. Why the change?
`torch.channels_last_3d ` could be used for 5-dim tensor and some 3d conv layers'. This PR tries to automatically handle
1. which data should be transformed to channels_last/channels_last_3d/remains original
2. which of channels_last/channels_last_3d should be used on users' model

### 2. User API changes
nothing, what's new please find it in ut.

### 3. Summary of the change 
1. change `channels_last_available` to be a string list to support 3d
2. related change to transform data/model

### 4. How to test?
- [ ] Unit test

### CODE REVIEW

**Comment by TheaperDeng** [comment]
@ACupofAir please also have a look

**Review by ACupofAir — APPROVED** [review]
LGTM

**Review by MeouSker77 — APPROVED** [review]
LGTM

File: python/nano/src/bigdl/nano/deps/ipex/ipex_inference_model.py [hunk 1]
```diff
@@ -150,17 +156,19 @@ def on_forward_start(self, inputs):
...
+            if not self.channels_last_available:
                 self.channels_last_available = generate_channels_last_available(inputs)
-                for idx, input in enumerate(inputs):
-                    if self.channels_last_available[idx]:
-                        input.to(memory_format=torch.channels_last)
+
+            # change the data to suitable mem format
+            for idx, input in enumerate(inputs):
+                if self.channels_last_available[idx] == "channels_last":
+                    input.to(memory_format=torch.channels_last)
```
Comment (rnwang04) [inline python/nano/src/bigdl/nano/deps/ipex/ipex_inference_model.py:168]: Not sure whether this way actually works

File: python/nano/test/pytorch/tests/inference/test_ipex_jit_inference.py [hunk 2]
```diff
@@ -143,6 +169,19 @@ def test_jit_channels_last_inference(self):
             load_model = InferenceOptimizer.load(tmp_dir_name, model)
             load_model(x1, x2, x3)
 
+    def test_jit_channels_last_3d_inference(self):
+        model = DummyModelWith3d()
+        x1 = torch.rand(32, 3, 3, 224, 224) # 5-dim input test
+        x2 = 3
+        jit_channels_last_model = InferenceOptimizer.trace(model, accelerator="jit", 
+                                                           use_ipex=False)
```
Comment (rnwang04) [inline python/nano/test/pytorch/tests/inference/test_ipex_jit_inference.py:177]: seems missing `channels_last=True`

File: python/nano/test/pytorch/tests/inference/test_ipex_jit_inference.py [hunk 3]
```diff
@@ -157,6 +196,19 @@ def test_ipex_jit_channels_last_inference(self):
             load_model = InferenceOptimizer.load(tmp_dir_name, model)
             load_model(x1, x2, x3)
 
+    def test_ipex_jit_channels_last_3d_inference(self):
+        model = DummyModelWith3d()
+        x1 = torch.rand(32, 3, 3, 224, 224) # 5-dim input test
+        x2 = 3
+        ipex_jit_channels_last_model = InferenceOptimizer.trace(model, accelerator="jit", 
+                                                                use_ipex=True)
```
Comment (rnwang04) [inline python/nano/test/pytorch/tests/inference/test_ipex_jit_inference.py:204]: seems missing `channels_last=True`

File: python/nano/test/pytorch/tests/inference/test_bf16_ipex.py [hunk 4]
```diff
@@ -298,6 +312,21 @@ def test_ipex_jit_inference_onednn(self):
             new_model(x)
             assert new_model.enable_onednn is True
 
+    def test_ipex_jit_channels_last_3d_inference(self):
+        model = DummyModelWith3d()
+        x1 = torch.rand(32, 3, 3, 224, 224) # 5-dim input test
+        x2 = 3
+        ipex_jit_channels_last_model = InferenceOptimizer.trace(model, accelerator="jit", 
+                                                                use_ipex=True, precision='bf16',
```
Comment (rnwang04) [inline python/nano/test/pytorch/tests/inference/test_bf16_ipex.py:320]: seems missing `channels_last=True`

**Review by rnwang04 — COMMENTED** [review]
others LGTM

File: python/nano/src/bigdl/nano/deps/ipex/ipex_inference_model.py [hunk 5]
```diff
@@ -150,17 +156,19 @@ def on_forward_start(self, inputs):
...
-                    if self.channels_last_available[idx]:
-                        input.to(memory_format=torch.channels_last)
+
+            # change the data to suitable mem format
+            for idx, input in enumerate(inputs):
+                if self.channels_last_available[idx] == "channels_last":
+                    input = input.to(memory_format=torch.channels_last)
+                if self.channels_last_available[idx] == "channels_last_3d":
+                    input = input.to(memory_format=torch.channels_last_3d)
+
```
Comment (rnwang04) [inline python/nano/src/bigdl/nano/deps/ipex/ipex_inference_model.py:171]: I am afraid the modification for input still does not work.
```python
def modify(al):
    for a in al:
        a += 1

sample = [1,2,3,4]
modify(sample)
print(sample) # [1,2,3,4]
```

**Comment by TheaperDeng** [comment]
ipex quantization model will be fixed next PR to avoid conflict since other PR is working on it.

**Review by rnwang04 — APPROVED** [review]
LGTM

**Pull request merged**